js/foam/browser/u2/StackView.js

```diff
@@ -0,0 +1,396 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+CLASS({
+  package: 'foam.browser.u2',
+  name: 'StackView',
+  extends: 'foam.u2.Element',
+
+  imports: [
+    'setTimeout',
+    'window'
+  ],
+
+  exports: [
+    'as stack'
+  ],
+
+  constants: {
+    VIEW_DESTROYED: ['view-destroyed']
+  },
+
+  properties: [
+    {
+      name: 'maxVisibleViews',
+      type: 'Int',
+      defaultValue: 2,
+      documentation: 'Limits the number of views on screen at a time.',
+    },
+    {
+      name: 'views_',
+      documentation: 'Internal array of child views.',
+      factory: function() {
+        return [];
+      }
+    },
+    {
+      name: 'visibleStart_',
+      documentation: 'The index of the leftmost visible element.',
+      defaultValue: 0
+    },
+    {
+      name: 'visibleEnd_',
+      documentation: 'The index of the rightmost visible element.',
+      defaultValue: 0
+    },
+    {
+      name: 'noDecoration',
+      type: 'Boolean',
+      documentation: 'Set true to hide panel edges.',
+      defaultValue: false
+    },
+    {
+      model_: 'foam.core.types.StringEnumProperty',
+      name: 'transition',
+      documentation: 'Transition to use when adding or removing a view.',
+      defaultValue: 'slide',
+      choices: ['slide', 'fade']
+    },
+  ],
+
+  methods: [
+    function init() {
+      this.SUPER();
+      this.window.addEventListener('resize', this.onResize);
+    },
+    {
+      name: 'pushView',
+      documentation: 'Default pushView that works at the top level. See ' +
+          '$$DOC{ref:".pushView_"} for details.',
+      code: function(view, hints) {
+        this.pushView_(-1, view, hints);
+      }
+    },
+    {
+      name: 'popView',
+      documentation: 'Default popView that works at the top level. See ' +
+          '$$DOC{ref:".popView_"} for details.',
+      code: function() {
+        this.popView_(0);
+      }
+    },
+    {
+      name: 'popChildViews',
+      documentation: 'Default popChildViews that works at the top level. See ' +
+          '$$DOC{ref:".popChildViews_"} for details.',
+      code: function() {
+        this.popView_(1);
+      }
+    },
+    function elementAnimationAdd_(e) {
+      if (this.transition === 'slide') {
+        e.style({
+          width: '0px',
+          left: this.id$el.offsetWidth,
```

Comment:
Do we need to access this.id$el.offsetWidth ?  What about something like left: 100%?  Or just pick a fixed big value.